modules/internal/ChapelLocale.chpl

```diff
@@ -186,7 +186,21 @@ module ChapelLocale {
     proc localeid : chpl_localeID_t return __primitive("_wide_get_locale", this);
 
     /*
-      Get the name of this locale.
+      Get the hostname of this locale.
+
+      :returns: the hostname of the compute node associated with the locale
+      :rtype: string
+    */
+    proc hostname: string {
+      extern proc chpl_nodeName(): c_string;
+      var name = chpl_nodeName():string;
```

Comment:
It looks like this would need an `on this` to me but I may be missing something. Does this work (or is it supposed to) for the following?

```chapel
on Locales[0] do writeln(Locales[1].hostname);
```

Reply:
Excellent point! (sheesh...)